src/__tests__/config.js

```diff
@@ -0,0 +1,60 @@
+import {configure, getConfig} from '../config'
+
+describe('configuration API', () => {
+  let originalConfig
+  beforeEach(() => {
+    // Grab the existing configuration so we can restore
+    // it at the end of the test
+    configure(existingConfig => {
+      originalConfig = existingConfig
+      // Don't change the existing config
+      return {}
+    })
+  })
+  afterEach(() => {
+    configure(originalConfig)
+  })
+
+  beforeEach(() => {
+    configure({other: 123})
+  })
+
+  describe('getConfig', () => {
+    test('returns existing configuration', () => {
+      const conf = getConfig()
+      expect(conf.testIdAttribute).toEqual('data-testid')
+    })
+  })
+
+  describe('configure', () => {
+    test('merges a delta rather than replacing the whole config', () => {
```

Comment:
Hmmm... I think the fact that you can _add_ random properties to the configuration is an implementation detail and not necessarily a feature we care to support. By that I mean, if we change the implementation in the future to only allow you to configure keys of things we support then I don't want to have to bother with fixing these tests.

So if we could make only assertions for the features we care to support that would be great.

More specifically, this test is fine, so long as we change the assertion to:

```javascript
expect(conf).toMatchObject({testIdAttribute: 'data-testid'})
```

The fact that `newKey` exists as well is a coincidence and not necessarily something we need to assert in our tests.